### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by anqit** [flyerhq/flutter_chat_ui#194] [Ideas & Requests] [answered]
I'm thinking that when conversations get long enough, there is no need to keep older messages in memory. So my thought is that if we could do something like only keeping n messages in memory unless the user scrolls up high enough, at which point we could make a server request for the previous messages, something like `onEndReached`, but `onBeginningReached` instead?

**Comment by demchenkoalex** [comment]
Oooh not sure about that one, this is more suitable for the complete solution with backend and stuff, will be hard to make it reusable and w/o specific backend support.

**Comment by anqit** [comment]
Fair enough. If I do get to implementing something along these lines, I will shoot for a generic enough approach to hopefully be able to contribute to the library.

**Reply by demchenkoalex** [reply]
Sure! we can also try to hide it under a boolean property so it can be disabled if does not work for someone. Thanks!

**Comment by demchenkoalex** [comment, marked as answer]
closing in favour of https://github.com/flyerhq/flutter_chat_ui/discussions/657 - more details
